Trappist/src/Promact.Trappist.Core/Controllers/TestController.cs

```diff
@@ -52,5 +53,40 @@ public TestsController(ITestsRepository testRepository)
             var tests = await _testRepository.GetAllTestsAsync();
             return Ok(tests);
         }
+
+        /// <summary>
+        /// Gets the Settings saved for a particular Test
+        /// </summary>
+        /// <param name="id">The parameter "id" is used to get the Settings of a Test by its Id</param>
+        /// <returns>Settings saved for the selected Test</returns>
+        [HttpGet("{id}")]
+        public async Task<IActionResult> GetTestSettings([FromRoute] int id)
+        {
+            var settings = await _testRepository.GetTestSettingsAsync(id);
+            if (settings == null)
+                return NotFound();
+            else
+                return Ok(settings);
+        }
+
+        /// <summary>
+        /// Updates the changes made to the settings of a Test
+        /// </summary>
+        /// <param name="id">The parameter "id" is used to access the Settings of that Test</param>
+        /// <param name="testSettingsAC">The parameter "testSettingsAC" is an object of TestSettingsAC</param>
+        /// <returns>Updated Settings of that Test</returns>     
+        [HttpPut("{id}")]
+        public async Task<IActionResult> UpdateTestSettings([FromRoute] int id, [FromBody] TestSettingsAC testSettingsAC)
+        {
+            if (!await _testRepository.TestSettingsExists(id))
```

Comment:
According to me, it's not TestSettingsExisits but you should check for Test Exists. You are not doing any special operation to check for TestSettingsExists. You are just checking Test Exists or not. So that you can do by Find method. A whole new method is not required.